pytorch_lightning/accelerators/accelerator.py

```diff
@@ -44,15 +45,29 @@ class Accelerator:
     One to handle differences from the training routine and one to handle different precisions.
     """
 
-    def __init__(self, precision_plugin: PrecisionPlugin, training_type_plugin: TrainingTypePlugin) -> None:
+    def __init__(
+        self, training_type_plugin: TrainingTypePlugin, precision_plugin: Optional[PrecisionPlugin] = None
+    ) -> None:
         """
         Args:
-            precision_plugin: the plugin to handle precision-specific parts
             training_type_plugin: the plugin to handle different training routines
+            precision_plugin: the plugin to handle precision-specific parts
+
+                .. deprecated:: v1.6
+                    The ``precision_plugin`` parameter has been deprecated in v1.6 and will be removed in 1.7.
```

Comment:
```suggestion
                    The ``precision_plugin`` parameter has been deprecated in v1.6 and will be removed in 1.8.
```
Should we remove it for v1.8 release instead?

Reply:
I'm not sure. A few thoughts:

- I'm not even sure we can uphold this deprecation/backward compatibility if we go all the way with https://docs.google.com/document/d/1E5t8auWf5DrNHzutvMmrJC_KqBqVyZuYX0thra69Ad8/edit#heading=h.nehmb825f40s
- We want accelerator/plugins to be stable at 1.6 - so we might as well introduce larger breaking changes now 
- So far we have always chosen a shorter deprecation of only one release for all plugin/accelerator related items.

RFC @ananthsub @four4fish

Reply:
Yeah I agree, in step 3 move accelerator into TTP, there is no way we can keep backwards compatibility. Precision plugin has to be removed then. 

> We want accelerator/plugins to be stable at 1.6 - so we might as well introduce larger breaking changes now

Agree! how about "this is deprecated and will be removed soon"?

Reply:
The precision plugin would be removed from here once we invert the Strategy/Accelerator hierarchy. @four4fish I don't think the deprecation warning would last through that refactor.

If that's the case, then we don't need the warning here. We can add the basic support in the interim for people on master, but the Accelerator shouldn't accept a precision plugin when 1.6 is released, since the accelerator won't have a reference to the strategy/training type plugin by then.

Reply:
Agree, will update.
I think as soon as the step 3: moving accelerator start, accelerator won't accept precision plugin anymore.